api/src/main/java/org/pmiops/workbench/billing/BillingAlertsService.java

```diff
@@ -20,49 +26,77 @@
   private final BigQueryService bigQueryService;
   private final NotificationService notificationService;
   private final WorkspaceDao workspaceDao;
+  private final WorkspaceFreeTierUsageDao workspaceFreeTierUsageDao;
   private final Provider<WorkbenchConfig> workbenchConfigProvider;
 
   @Autowired
   public BillingAlertsService(
       BigQueryService bigQueryService,
       NotificationService notificationService,
       WorkspaceDao workspaceDao,
+      WorkspaceFreeTierUsageDao workspaceFreeTierUsageDao,
       Provider<WorkbenchConfig> workbenchConfigProvider) {
     this.bigQueryService = bigQueryService;
     this.notificationService = notificationService;
     this.workspaceDao = workspaceDao;
+    this.workspaceFreeTierUsageDao = workspaceFreeTierUsageDao;
     this.workbenchConfigProvider = workbenchConfigProvider;
   }
 
   public void checkFreeTierBillingUsage() {
-    QueryJobConfiguration queryConfig =
+    final Map<Workspace, Double> workspaceCosts = getFreeTierWorkspaceCosts();
+
+    final Set<User> expiredCreditsUsers =
+        workspaceCosts.entrySet().stream()
+            .collect(
+                Collectors.groupingBy(
+                    e -> e.getKey().getCreator(), Collectors.summingDouble(Entry::getValue)))
+            .entrySet()
+            .stream()
```

Comment:
I'm not super happy about `collect()` and then `stream()` again, but I didn't see an obvious better solution.

Reply:
Could you do it with a:
```
.map(entry -> entry.setKey(entry.getKey().getCreator()); entry.setValue(_something_);
```

Reply:
Recognizing that it isn't that simple, but maybe calling a constructor or a duplicator for an entry in there.

Reply:
@als364 I need both `workspaceCosts` and `expiredCreditsUsers`.  Since the former contains everything needed for the latter, my preference is to get both of these via a single BigQuery query and this transformation.

Reply:
Here's a StackOverflow with ideas but none of them *quite* fit so I'm inclined to keep this version: https://stackoverflow.com/questions/44101297/java-stream-filter-sum-of-values